models/media.py

```diff
@@ -0,0 +1,267 @@
+        self.file = file
+
+        # Try to cast the file_type as a MediaType
+        try:
+            self.media_type = MediaType(file.content_type)
+        except Exception:
+            raise InvalidMediaTypeException(
+                "Sorry '{}' is not a supported format".format(
+                    file.content_type))
+
+        self.uuid = uuid.uuid4()
+
+    def set_exif(self):
+        image = Image.open(self.file)
+
+        try:
+            exif = {
+                ExifTags.TAGS[k]: v
+                for k, v in image._getexif().items()
+                if k in ExifTags.TAGS
+            }
+        except Exception:
+            return
+
+        exposure = exif.get("ExposureTime", None)
+
+        if exposure is not None:
+            self.shot_exposure = "{}/{}".format(exposure[0], exposure[1])
+
+        aperture = exif.get("FNumber", None)
+
+        if aperture is not None:
+            self.shot_aperture = "ƒ{}".format(round(aperture[0] / aperture[1], 1))
+
+        self.shot_speed = exif.get("ISOSpeedRatings", None)
+
+        self.shot_focal_length = str(exif.get("FocalLengthIn35mmFilm", None)) + "mm"
+
+        self.camera_make = exif.get("Make", None)
+        self.camera_model = exif.get("Model", None)
+
+        self.camera_make = EXIF_NAME_MAPS.get(self.camera_make, self.camera_make)
+        self.camera_model = EXIF_NAME_MAPS.get(self.camera_model, self.camera_model)
+
+        pass
+
+    def set_static_info(self):
+        image = Image.open(self.file)
+
+        self.width = image.width
+        self.height = image.height
+
+        self.aspect = str(round(self.width / self.height, 2))
+
+    def optimize(self):
+        if self.media_type in STATIC_MEDIA_TYPES:
+            self.set_static_info()
+            self.set_exif()
+
+            return self.optimize_static()
+
+        elif self.media_type is MediaType.GIF:
+            return self.optimize_gif()
+
+        else:
+            raise Exception("Called optimize on media that cannot be optimized")
+
+    def set_average_color(self, image):
+
+        def rgb_to_hex(rgb):
+            return '%02x%02x%02x' % rgb
+
+        try:
+            color_thief = ColorThiefFromImage(image)
+            avg_color = rgb_to_hex(color_thief.get_color(quality=1))
+        except Exception:
+            self.average_color = DEFAULT_AVG_COLOR
+
+            return DEFAULT_AVG_COLOR
+
+        self.average_color = avg_color
+
+        return avg_color
+
+    def optimize_gif(self):
+        # Save GIF to FS
+        raw_gif_file_name = str(self.uuid) + "." + self.media_type.ext()
+        self.file.save(raw_gif_file_name)
+
+        # Cast GIF to Clip
+        clip = mp.VideoFileClip(raw_gif_file_name)
+
+        temp_poster_file_name = str(self.uuid) + ".poster.jpeg"
+        clip.save_frame(temp_poster_file_name, t=0)
+        poster = Image.open(temp_poster_file_name)
+
+        self.set_average_color(poster)
+
+        self.width = clip.w
```

Comment:
This could be done from the poster and could be done with set_info if you pass the image in